File: app/src/main/java/com/lingoswap/domain/usecase/auth/ForgotPasswordUseCase.java
package com.lingoswap.domain.usecase.auth;

import com.lingoswap.data.model.ApiResponse;
import com.lingoswap.data.model.ForgotPasswordRequest;
import com.lingoswap.data.repository.RepositoryCallback;
import com.lingoswap.domain.repository.AuthRepository;

import javax.inject.Inject;

public class ForgotPasswordUseCase {
    private final AuthRepository repository;

    @Inject
    public ForgotPasswordUseCase(AuthRepository repository) {
        this.repository = repository;
    }

    public void execute(String email, RepositoryCallback<ApiResponse> callback) {
        // AuthRepository interface needs forgotPassword method
    }
}

File: app/src/main/java/com/lingoswap/domain/usecase/auth/ForgotPasswordUseCase.kt
package com.lingoswap.domain.usecase.auth

import com.lingoswap.domain.repository.AuthRepository
import com.lingoswap.utils.Resource
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class ForgotPasswordUseCase @Inject constructor(
    private val repository: AuthRepository
) {
    suspend operator fun invoke(email: String): Flow<Resource<Unit>> {
        return repository.forgotPassword(email)
    }
}
